onager/backends/slurm.py

```diff
@@ -1,15 +1,47 @@
 from datetime import timedelta
 import os
+import math
 
 from ._backend import Backend
-
+from ..utils import expand_ids
 
 class SlurmBackend(Backend):
     def __init__(self):
         super().__init__()
         self.name = 'slurm'
         self.task_id_var = r'$SLURM_ARRAY_TASK_ID'
 
+
+    def _get_body(self, tasks_file, args):
+        """
+        A bit complicated. It may have a bunch of concurrent processes.
+        """
```

Comment:
tasklist is all of the indices of jobs.json that you want to run this job. How we do this is, we make a bash array of all of the indices we want to run for this job, then depending on the slurm ID choose the right ones to run in parallel.

There's an edge case we deal with, for example if tasklist=1,2,3 and tasks_per_job=2. In that case, it will make 2 jobs, but the second one will only run task-id 3 (because there's no fourth entry). This script would try and access index 4 of the bash array as it is now, which would break the job launching by running onager.worker without the right arguments. So we pad the array with __filler__ and have onager.worker know to skip these.